internal/dao/rbac_policy.go

```diff
@@ -159,18 +160,30 @@ func (p *Policy) fetchRoleBindingSubjects(kind, name string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	ns, n := client.Namespaced(name)
 	ss := make([]string, 0, len(rbs))
 	for _, rb := range rbs {
 		for _, s := range rb.Subjects {
-			if s.Kind == kind && s.Name == name {
-				ss = append(ss, rb.RoleRef.Kind+":"+rb.Name)
+			if isSameSubject(kind, ns, n, s) {
+				ss = append(ss, rb.RoleRef.Kind+":"+rb.RoleRef.Name)
 			}
 		}
 	}
 
 	return ss, nil
 }
 
+// isSameSubject verifies if the incoming type name and namespace match a subject from a
+// cluster/roleBinding. A ServiceAccount will always have a namespace and needs to be validated to ensure
+// we don't display permissions for a ServiceAccount with the same name in a different namespace
+func isSameSubject(kind, namespace, name string, subject rbacv1.Subject) bool {
```

Comment:
I think we should pass in a pointer to the subject here. Also pass in the subject fqn name and break it down in the class via client.Namespaced. Lastly would be great to have a test on this!
```go
func hasSameSubject(s *rbacv1.Subject, kind, name string) bool {...}
```

Reply:
I updated to use a pointer but in the loop I have to make a new reference to the subject so it does add some noise. 
For the fqn I don't think that's needed since I have a subject with the info already broken down. If you are referring to passing in the fqn for the `KeySubjectName` I break that down outside the loop so I don't have to process it for every subject.